### DISCUSSION TITLE
widget.List behaviour on mobile + another callback

### DISCUSSION THREAD

**Discussion by blaize** [fyne-io/fyne#4087] [Q&A]
Hello,
I love working with lists. There are 2 things really missing for my app. Before start implementing it and make a PR, I wanted to have your review.
First, I noticed the behaviour of selecting item in list on android is not consistant with selecting item in fyne app. In android, selecting item is done by long press. Taping an item run another action, that can be called "open" the item (for more info on it, display larger, etc...)
That lead to the second point, I want another callback, say "OnOpen()" that be triggered when simple tap occur on item.
So, to summary:
1/ Select is done by simple click on desktop, long press on mobile
2/ Open is done by double click on desktop, simple tap on mobile.

Opening item is also very handy and clean the UI because you don't need to add a button on each row anymore, or to look for a button to open after selecting an item.
Yes I can implement a label with the event override but if the row is more than just a label, it's not intuitive to open.

What do you think ?

**Comment by andydotxyz** [comment]
The premise here isn't right - maybe was determined by looking at fyne_demo not the API itself?
Anyhow, a list is selected by single tap on all platforms. So the call to Selected is already what you want (fires on tap) - what happens next is up to the application.

**Reply by blaize** [reply]
I want 2 differents events. One for selecting, one for opening.
Actually I can't open an item with just one callback. I have to add a button on each row, that mean more complexe UI and more element to display make the list slower to scroll.
Or I open a list element with OnSelect event but I can't just select the element then.

Sorry for my bad english, It's what I mean : "a list is selected by single tap on all platforms"
This should not be the case.
On android, every other app use long press to select item. I mean, when you are used to work with your phone, you expect an app to be consistent with your habits. Simple tapping on an item make another action, not select.
That's also true on desktop where double clicking generally "open" things and simple click select things.

**Reply by andydotxyz** [reply]
List is a data display not a navigation widget. "Opening" things on single tap can be done, though it is not part of the API.

You can add long tap and double tap to your items simply by adding that to the objects you put in the list, it does not need to be part of list.

> That's also true on desktop where double clicking generally "open" things and simple click select things.

Double click should never be your default action, it is not hinted in the UI. Windows file manager has taught many generations bad habits!

**Reply by andydotxyz** [reply]
> On android, every other app use long press to select item. 

I was unaware of this. On the iOS platform and apps that work across them both it is more common that it pops up context menus or sets things to an edit mode (more like right click than left click on desktop). This is the pattern we follow with TappedSecondary.

**Reply by blaize** [reply]
Ok I got it.
Thanks for the review, I will keep it as is.
